src/System.Management.Automation/engine/Interop/Windows/CreateFile.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Management.Automation;
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static unsafe partial class Windows
+    {
+        // dwDesiredAccess of CreateFile
+        [Flags]
+        internal enum FileDesiredAccess : uint
+        {
+            GenericZero = 0,
+            GenericRead = 0x80000000,
+            GenericWrite = 0x40000000,
+            GenericExecute = 0x20000000,
+            GenericAll = 0x10000000,
+        }
+
+        // dwFlagsAndAttributes
+        [Flags]
+        internal enum FileAttributes : uint
```

Comment:
If we are trying to unify this, isn't it ideal to do it properly first then migrate the code.

Reply:
> Personally I recommend having FileAttributes as one parameter and another custom enum just for the flags and in the CreateFile call bitor them together

Since this is low-level internal code for P/Invoke, there's no particular reason to complicate matters by using the public FileAttributes Enum.

My preference is to stay as close a possible to the underlying API, see: https://github.com/dotnet/pinvoke/blob/main/src/Kernel32/Kernel32+CreateFileFlags.cs